packages/apps/human-app/frontend/src/modules/worker/profile/components/staking-info.tsx

```diff
@@ -0,0 +1,141 @@
+import { Button, IconButton, Skeleton, Stack, Typography } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+import { Chip } from '@/shared/components/ui/chip';
+import { env } from '@/shared/env';
+import { useAddApiKeyModal } from '../hooks/use-api-key-modals';
+import { ApiKeyData } from './';
+import { useGetStakingSummary } from '../../hooks/use-staking';
+import { RefreshIcon } from '@/shared/components/ui/icons';
+import { useGetExchangeApiKeys } from '../../hooks/use-exchange-api-keys';
+import { useAuthenticatedUser } from '@/modules/auth/hooks/use-authenticated-user';
+import { useEffect, useRef } from 'react';
+import { useAccessTokenRefresh } from '@/api/hooks/use-access-token-refresh';
+
+export function StakingInfo() {
+  const { user } = useAuthenticatedUser();
+  const { refreshAccessTokenAsync } = useAccessTokenRefresh();
+  const { t } = useTranslation();
+  const hasAttemptedRefresh = useRef(false);
+
+  const { openModal: openAddApiKeyModal } = useAddApiKeyModal();
+  const { data: exchangeApiKeyData, isLoading: isExchangeApiKeyLoading } =
+    useGetExchangeApiKeys();
+  const {
+    data: stakingSummary,
+    isLoading,
+    isError,
+    refetch,
+    isRefetching,
+  } = useGetStakingSummary();
+
+  const isConnectButtonDisabled =
+    !!exchangeApiKeyData?.exchange_name || isExchangeApiKeyLoading;
+
+  const stakedAmount =
+    Number(stakingSummary?.on_chain_stake || 0) +
+    Number(stakingSummary?.exchange_stake || 0);
+
+  const isStaked =
+    stakedAmount >= Number(stakingSummary?.min_threshold || '1000');
+
+  const isStakingError =
+    !!stakingSummary?.on_chain_error ||
+    !!stakingSummary?.exchange_error ||
+    isError;
+
+  useEffect(() => {
+    if (isRefetching || isLoading) return;
+
+    if (isStaked !== user.is_stake_eligible) {
+      if (!hasAttemptedRefresh.current) {
+        hasAttemptedRefresh.current = true;
+        refreshAccessTokenAsync({ authType: 'web2' });
```

Comment:
Doesn't affect the result, but saves some mind-fuel :)
```suggestion
        return refreshAccessTokenAsync({ authType: 'web2' });
```